geth/account/accounts_test.go

```diff
@@ -103,9 +107,381 @@ func TestVerifyAccountPasswordWithAccountBeforeEIP55(t *testing.T) {
+	nodeManager := newMockNodeManager(t)
+	accManager := NewManager(nodeManager)
+
+	keyStoreDir, err := ioutil.TempDir(os.TempDir(), "accounts")
+	require.NoError(t, err)
+	keyStore := keystore.NewKeyStore(keyStoreDir, keystore.LightScryptN, keystore.LightScryptP)
+	defer os.RemoveAll(keyStoreDir) //nolint: errcheck
+
+	testPassword := "test-password"
+
+	// Initial test - create test account
+	nodeManager.EXPECT().AccountKeyStore().Return(keyStore, nil)
+	addr, pubKey, mnemonic, err := accManager.CreateAccount(testPassword)
+	require.NoError(t, err)
+	require.NotNil(t, addr)
+	require.NotNil(t, pubKey)
+	require.NotNil(t, mnemonic)
+
+	s := &ManagerTestSuite{
+		testAccount: testAccount{
+			"test-password",
+			addr,
+			pubKey,
+			mnemonic,
+		},
+		nodeManager:    nodeManager,
+		accManager:     accManager,
+		keyStore:       keyStore,
+		shh:            whisper.New(nil),
+		gethAccManager: accounts.NewManager(),
+	}
+
+	suite.Run(t, s)
+}
+
+func newMockNodeManager(t *testing.T) *common.MockNodeManager {
+	ctrl := gomock.NewController(t)
+	return common.NewMockNodeManager(ctrl)
+}
+
+type ManagerTestSuite struct {
+	suite.Suite
+	testAccount
+	nodeManager    *common.MockNodeManager
+	accManager     *Manager
+	keyStore       *keystore.KeyStore
+	shh            *whisper.Whisper
+	gethAccManager *accounts.Manager
+}
+
+type testAccount struct {
+	password string
+	address  string
+	pubKey   string
+	mnemonic string
+}
+
+// reinitMock is for reassigning a new mock node manager to account manager.
+// Stating the amount of times for mock calls kills the flexibility for
+// development so this is a good workaround to use with EXPECT().AnyTimes()
+func (s *ManagerTestSuite) reinitMock() {
+	s.nodeManager = newMockNodeManager(s.T())
+	s.accManager.nodeManager = s.nodeManager
+}
+
+func (s *ManagerTestSuite) TestCreateAndRecoverAccount() {
+	s.reinitMock()
+
+	// Don't fail on empty password
+	s.nodeManager.EXPECT().AccountKeyStore().Return(s.keyStore, nil)
+	_, _, _, err := s.accManager.CreateAccount(s.password)
+	s.NoError(err)
+
+	// Recover the account using the mnemonic seed and the password
+	s.nodeManager.EXPECT().AccountKeyStore().Return(s.keyStore, nil)
+	addr, pubKey, err := s.accManager.RecoverAccount(s.password, s.mnemonic)
+	s.NoError(err)
+	s.Equal(s.address, addr)
+	s.Equal(s.pubKey, pubKey)
+
+	s.nodeManager.EXPECT().AccountKeyStore().Return(nil, testErrKeyStore)
+	_, _, _, err = s.accManager.CreateAccount(s.password)
+	s.Equal(err, testErrKeyStore)
+
+	s.nodeManager.EXPECT().AccountKeyStore().Return(nil, testErrKeyStore)
+	_, _, err = s.accManager.RecoverAccount(s.password, s.mnemonic)
+	s.Equal(err, testErrKeyStore)
+}
+
+func (s *ManagerTestSuite) TestSelectAccount() {
+	s.reinitMock()
+
+	testCases := []struct {
+		name                  string
+		accountKeyStoreReturn []interface{}
+		whisperServiceReturn  []interface{}
+		address               string
+		password              string
+		fail                  bool
```

Comment:
Alternatively, you can replace `fail bool` with `err error` and check if the returned error is equal. In case of a `nil`, you can check there is no error.

Reply:
Makes sense to me. I'll do a rewriting for all regarding this.